raft/server.cc

```diff
@@ -987,8 +997,8 @@ future<> server_impl::applier_fiber() {
         while (true) {
             auto v = co_await _apply_entries.pop_eventually();
 
-            if (std::holds_alternative<std::vector<log_entry_ptr>>(v)) {
-                auto& batch = std::get<0>(v);
+            if (auto* batch_opt = std::get_if<std::vector<log_entry_ptr>>(&v)) {
+                auto& batch = *batch_opt;
```

Comment:
Why is this change?

Reply:
Using `get_if` is better than `holds_alternative`+`get`, because the latter performs two checks, one of them is redundant.

I used the better option - `get_if` - in the `else` branch. Then the code became inconsistent with this first `if`, so I changed this first `if` as well to make the code consistent.

Reply:
I do not think that `get<0>` performs any checks at runtime.

Reply:
I do not mind using visit but as a separate patch.

Reply:
>  I do not think that get<0> performs any checks at runtime.

It does: https://en.cppreference.com/w/cpp/utility/variant/get

> Index-based value accessor: If v.index() == I, returns a reference to the value stored in v. Otherwise, throws [std::bad_variant_access](https://en.cppreference.com/w/cpp/utility/variant/bad_variant_access). The call is ill-formed if I is not a valid index in the variant.

Reply:
OK, that it is still same to `ifs`. Either `holds_alternative` returns bool and we check it and then `get<>` does one more `if`. Or `get_if<>` does a check and then there is one more null comparison in the `if`. The point is if you want to change a style better to do it in a separate patch. Here just add another `if` and then you can move the whole thing to `visit`.

Reply:
I would guess that the variant checks are some type of virtual calls, as std variants are implemented in terms of virtual calls, much more expensive than comparing a pointer to `null`.

Anyway, I will extract to separate patch.

Reply:
Yes, variant has function table. Do we really care here?

Reply:
Probably not.